### PR TITLE
Add One-Click Deploy to Heroku Button

### PR DESCRIPTION
**Pull request by rrao24**
This PR introduces a 'Deploy to Heroku' button in the README to make it simpler for users to deploy this application to heroku. The file app.json was also added to the project as it is necessary for the heroku button.

I had previously created a PR for these changes, but I closed it as the commits were not squashed properly. This PR fixed that.

Thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ Navigation: [Website][1] | **Server repository** | [Client repository][2]
 [![Build Status](https://travis-ci.org/Zarel/Pokemon-Showdown.svg)](https://travis-ci.org/Zarel/Pokemon-Showdown)
 [![Dependency Status](https://david-dm.org/zarel/Pokemon-Showdown.svg)](https://david-dm.org/zarel/Pokemon-Showdown)
 [![devDependency Status](https://david-dm.org/zarel/Pokemon-Showdown/dev-status.svg)](https://david-dm.org/zarel/Pokemon-Showdown#info=devDependencies)
-
+[![Deploy to Heroku](https://www.herokucdn.com/deploy/button.png)](https://heroku.com/deploy)
```
Comment (Zarel) [inline README.md]: This doesn't seem to be enough to deploy PS specifically, unless it actually reads the referer? Even so I think HTTPS links cut off the referer at the domain.

**Comment by darkcl** [comment]
config/config.js is in .gitignore, the deploy button won't work properly

**Comment by Zarel** [comment]
There's a stub config file in `config/config-example.js` which is copied over on launch. It should be fine.

**Comment by Zarel** [comment]
On the other hand, closing since this code appears to not work and is unmaintained.

**Pull request closed without merging**